features/ultils/credentials.ts

```diff
@@ -0,0 +1,152 @@
+import { KintoneRestAPIClient } from "@kintone/rest-api-client";
+import type { Credential, Permission } from "./types";
+import fs from "fs";
+import path from "path";
+import Ajv from "ajv";
+import type { ErrorObject } from "ajv";
+
+type Record = {
+  key: {
+    type: "SINGLE_LINE_TEXT";
+    value: string;
+  };
+  app_id: {
+    type: "SINGLE_LINE_TEXT";
+    value: string;
+  };
+  api_tokens: {
+    type: "SUBTABLE";
+    value: Array<{
+      id: string;
+      value: {
+        token: {
+          type: "SINGLE_LINE_TEXT";
+          value: string;
+        };
+        permissions: {
+          type: "CHECK_BOX";
+          value: Permission[];
+        };
+      };
+    }>;
+  };
+};
+
+const e2eCredentialFileName = ".e2e-credentials.json";
+const e2eCredentialFilePath = path.join(
+  __dirname,
+  `../../${e2eCredentialFileName}`,
+);
+
+const isRunOnLocal = () => !process.env.CI;
+
+const isRunOnCI = () => !!process.env.CI;
+
+export const fetchCredentials: () => Promise<Credential[]> = async () => {
```

Comment:
How about renaming it to `load`, a more abstractive name?
This function reads/fetches credentials from fs/API.


```suggestion
export const loadCredentials: () => Promise<Credential[]> = async () => {
```